File: Internship Project/backend-project/src/main/java/com/mergentech/internship_project/dtos/requests/BasvuruFormRequest.java
package com.mergentech.internship_project.dtos.requests;

import com.mergentech.internship_project.enums.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class BasvuruFormRequest {
    private String adSoyad;
    private String email;
    private String telefon;
    private String universite;
    private String bolum;
    private Siniflar sinif;
    private String linkedinLink;
    private String githubLink;
    private String cvLink;
    private List<DepartmanlarEnum> departman;
    private List<DillerEnum> progDil;
    private List<DepartmanlarEnum> departmanlar;
    private List<ProgDilRequest> diller;
    private Tarihler tarih;
    private String digerDep;
}

File: Internship Project/frontend/frontend-project/src/app/Request/BasvuruFormRequest.ts
import {Departmanlar} from "../enum/departmanlar.enums";

export class BasvuruFormRequest {
  id!: number;
  adSoyad!: null;
  email!: null;
  telefon!: null;
  universite!: null;
  bolum!: null;
  sinif: any;
  linkedinLink!: null;
  githubLink!: null;
  cvLink!: null;
  departmanlar: Departmanlar[]=[];
  dil!: any[];
  tarih: any;
  accepted: null | undefined;
  accepted2: null | undefined;
}
